### ISSUE TITLE
The Twisted Coding Standard should specify whether @since tags are x.y or x.y.z

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7765]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7765|
|Type|enhancement|
|Created|2015-02-02 22:30:43Z|
|Branch|https://github.com/twisted/twisted/tree/since-tags-docs-7765|

In the 15.0 release, I changed all instances of 14.1.0 to 15.0.0. This caught the deprecation warnings (which format to x.y.z) and SOME @since tags in documentation/docstrings, but not all, as some had `@since: 14.1` instead of `@since: 14.1.0`.

Both deprecation warnings and new features work on an API-level basis (introduced in x+1.y.z, x.y+1.z, but NOT x.y.z+1). This means that x.y should be the format that `@since:` uses.

The coding standard should specify which is correct, for future reference.

<details><summary>Searchable metadata</summary>

```
trac-id__7765 7765
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__ 
branch__branches_since_tags_docs_7765 branches/since-tags-docs-7765
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1422916243317072 1422916243317072
changetime__1423142698855957 1423142698855957
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [43810]) Branching to since-tags-docs-7765.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Changed it to specify x.y.z. Maybe see what core contributors think?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @hawkowl|
|-|-|

New features aren't allowed in a micro release.  The only thing you can ever write is "@since: x.y.0".  Writing this will just confuse people into thinking you might sometimes see "@since: x.y.3" otherwise why specify the extra `.0`?

This should be "@since: x.y".  (And it should be added automatically at release time but that's a different issue.)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Yeah, that makes sense. It's a bit weird that deprecation notices, which also can't be in a micro release, have the micro release as part of the warning message. I guess that's not as fixable.

I'll change that.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl removed owner|
|-|-|

Thanks for the review, exarkun.

I've made the necessary change, and I'll file a ticket to change the instances of x.y.z that were in the release to be the consistent x.y.

Putting up for review.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl commented|
|-|-|

(In [43825]) should be x.y (review comment by exarkun, refs [#7765](https://github.com/twisted/twisted/issues/7765))

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph commented|
|-|-|

Just forced the build to make sure the docs build OK.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_9"></a>|@glyph set owner to @hawkowl|
|-|-|

OK, docs seem to build fine, prose appears accurate; please land.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl set status to `closed`|
|-|-|

(In [43835]) Merge since-tags-docs-7765: The Twisted Coding Standard should specify that @since tags are x.y

Author: hawkowl
Reviewers: exarkun, glyph
Fixes: [#7765](https://github.com/twisted/twisted/issues/7765)
